### ISSUE TITLE
Industrialize the inference service(s)

### ISSUE THREAD

**Issue by Almarch** [Almarch/Pokedex#2]
Switch to vLLM for the LLM, and use a distinct service for embedding (fastapi + sentence transformer ?)

**Comment by Almarch** [comment]
This has been explored with vllm in: https://github.com/Almarch/pokedex/tree/feature/vllm

pros:
- reranking score from llm logits
- allow asymmetric embedding
- batch inference ?

cons:
- complexity overhead


maybe llama.cpp should be explored as well (especially for typed generation ?) - or straight transformers.
